consensus/core/src/dag_state.rs

```diff
@@ -103,45 +103,45 @@ impl DagState {
-                let mut committed_rounds = commit_info.committed_rounds;
-                let last_commit = last_commit
-                    .as_ref()
-                    .expect("There exists commit info, so the last commit should exist as well.");
-
-                if last_commit.index() > commit_ref.index {
-                    let committed_blocks = store
-                        .scan_commits(((commit_ref.index + 1)..last_commit.index() + 1).into())
-                        .unwrap_or_else(|e| panic!("Failed to read from storage: {:?}", e))
-                        .iter()
-                        .flat_map(|commit| {
-                            let committed_subdag =
-                                load_committed_subdag_from_store(store.as_ref(), commit.clone());
-                            if context
-                                .protocol_config
-                                .mysticeti_leader_scoring_and_schedule()
-                            {
-                                unscored_committed_subdags.push(committed_subdag.clone());
-                            }
-                            committed_subdag.blocks
-                        })
-                        .collect::<Vec<_>>();
-
-                    for block in committed_blocks {
-                        committed_rounds[block.author()] =
-                            max(committed_rounds[block.author()], block.round());
+                (commit_info.committed_rounds, commit_ref.index + 1)
+            } else {
+                (vec![0; num_authorities], GENESIS_COMMIT_INDEX + 1)
+            };
+
+        if let Some(last_commit) = last_commit.as_ref() {
+            let committed_blocks = store
+                .scan_commits((commit_recovery_start_index..last_commit.index() + 1).into())
+                .unwrap_or_else(|e| panic!("Failed to read from storage: {:?}", e))
+                .iter()
+                .flat_map(|commit| {
+                    let committed_subdag =
+                        load_committed_subdag_from_store(store.as_ref(), commit.clone());
+                    if context
+                        .protocol_config
+                        .mysticeti_leader_scoring_and_schedule()
+                    {
+                        unscored_committed_subdags.push(committed_subdag.clone());
                     }
-                }
+                    committed_subdag.blocks
+                })
+                .collect::<Vec<_>>();
 
-                committed_rounds
-            } else {
-                vec![0; num_authorities]
+            for block in committed_blocks {
+                last_committed_rounds[block.author()] =
+                    max(last_committed_rounds[block.author()], block.round());
             }
-        };
+        }
+
+        tracing::info!(
+            "DagState was initialized with the following state: \
+            {last_commit:?}; {last_committed_rounds:?}; {} unscored_committed_subdags;",
```

Comment:
Maybe only print the leader among `unscored_committed_subdags`. Otherwise the log message will be truncated.

Reply:
I am currently only printing the count of the subdags, but if you want me include the leaders as well I can.

Reply:
It is fine then.